include/vga.h

```diff
@@ -257,6 +257,8 @@ struct VgaDraw {
 	Bitu lines_scaled           = 0;
 	Bitu split_line             = 0;
 
+	bool is_double_scanning = false;
+
```

Comment:
Also have `double_scanning_enabled = false` in the same `vga.draw.` struct on line 294.  There's probably a good reason for these (maybe one is existing and the other is the new/changing state?).

Also have `video_mode.is_double_scanned_mode`, in a different struct - but maybe it could share the same name (if the meaning is identical?)

Reply:
These are three related but different things. All three are needed, they serve different purposes. In order:

- `VgaDraw.double_scanning_enabled` -- This existing flag tells us whether double-scanning is allowed or not. The regular VGA shaders allow double-scanning, in which case this is true. But we can force single-scanning in the arcade shaders with `#pragma force_single_scan`; then the flag is false. So this is basically a configuration option for the renderer.
- `VgaDraw.is_double_scanning` -- I introduced this now, this reflects actual reality. Are we in double-scanned rendering mode, yes or no? If `double_scanning_enabled` is false, this can only be false, of course, as double-scanning is disabled.
- `video_mode.is_double_scanned_mode` -- Regardless of whether double-scanning is currently enabled via `double_scanning_enabled` or not, this flag represents in the VideoMode struct whether the given mode is a double-scanned mode. This info is needed in the upscaler, and maybe in the shader switcher too.

Happy to change the names if you have improvement ideas. Maybe `double_scanning_enabled` could be called `double_scanning_allowed` (enabled can be misleading). But the other two I think are fine.

Reply:
Thanks for this explanation!

Yes, that rename to `_allowed` looking like a nice improvement.

I would also say your bullet comments here could be pasted straight into the structs just above each of these three variables (good IDEs will expand the preceding comment block, so would be nice to have your explanations always in the code :-)

Reply:
Yeah I think I've documented some of these structs/flags already, but I'll double check and add the missing descriptions where needed. Using "allowed" is definitely an improvement, agree (just read my above explanation, it just "begs" for allowed in the name 😆).